### PR TITLE
[CLN] remove now-unnecessary td.skip_if_no(pathlib)

### LINKED ISSUE #30375: Is `@td.skip_if_no("pathlib")` necessary?
Admittely this is very minor, but 
there's a few parts in the codebase (e.g. `pandas/tests/io/test_common.py`) where there is
```
@td.skip_if_no("pathlib")
```
As of `v1.0`, the minimum Python version supported will be 3.6.1, so `pathlib` will always be present as part of the standard library.

### PR DESCRIPTION
**Pull request by MarcoGorelli**
- [ ] closes #30375 
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/io/sas/test_sas7bdat.py [hunk 1]
```diff
@@ -68,7 +68,6 @@ def test_from_iterator(self):
                 tm.assert_frame_equal(df, df0.iloc[2:5, :])
                 rdr.close()
 
-    @td.skip_if_no("pathlib")
     def test_path_pathlib(self):
         from pathlib import Path
```
Comment (simonjayhawkins) [inline pandas/tests/io/sas/test_sas7bdat.py]: can move import to top?

File: pandas/tests/io/test_common.py [hunk 2]
```diff
@@ -73,7 +73,6 @@ def test_expand_user_normal_path(self):
         assert expanded_name == filename
         assert os.path.expanduser(filename) == expanded_name
 
-    @td.skip_if_no("pathlib")
```
Comment (simonjayhawkins) [inline pandas/tests/io/test_common.py]: L32-L37 can be removed and path_types updated?

File: pandas/tests/io/pytables/test_store.py [hunk 3]
```diff
@@ -4594,7 +4594,6 @@ def test_read_nokey_empty(self, setup_path):
             with pytest.raises(ValueError):
                 read_hdf(path)
 
-    @td.skip_if_no("pathlib")
```
Comment (simonjayhawkins) [inline pandas/tests/io/pytables/test_store.py]: import to top?

**Review by simonjayhawkins — COMMENTED** [review]
Thanks @MarcoGorelli for the PR. a couple of suggestions otherwise lgtm.

**Comment by MarcoGorelli** [comment]
@simonjayhawkins sure, done - thanks for your review, and sorry for not having picked up on that straight away

**Review by simonjayhawkins — APPROVED** [review]
@MarcoGorelli Thanks.

**Comment by jreback** [comment]
thanks @MarcoGorelli

**Pull request merged**